File: src/models/Restaurant.ts
import type { RestaurantKind, Contact, Image, PaymentMethod, Menu } from "~/models";

export interface Restaurant {
  id: number,
  title: string,
  latitude: number,
  longitude: number,
  area: string,
  address: string,
  opening: string,
  closing: string,
  kind: RestaurantKind,
  accessibility: boolean,
  wifi: boolean,
  shortDescription: string,
  description: string,
  access: string,
  operationalHours: string,
  contact: Contact,
  crousAndGo: string,
  album: Image | undefined,
  photo: Image,
  paymentMethods: Array<PaymentMethod>,
  menus: Array<Menu>
}

File: library/src/commonMain/kotlin/models/restaurant.kt
package models

import kotlinx.serialization.Serializable

@Serializable
data class Restaurant(
    val id: Int,
    val title: String,
    val latitude: Double,
    val longitude: Double,
    val area: String,
    val address: String,
    val opening: String,
    val closing: String,
    val kind: RestaurantKind,
    val accessibility: Boolean,
    val wifi: Boolean,
    val shortDescription: String,
    val description: String,
    val access: String,
    val operationalHours: String,
    val contact: Contact,
    val crousAndGo: String,
    val album: Image?,
    val photo: Image,
    val paymentMethods: List<PaymentMethod>,
    val menus: List<Menu>
)
